src/app/api/tasks/tasks.service.ts

```diff
@@ -264,6 +264,25 @@ export class TasksService extends BaseService {
     return await this.db.task.delete({ where: { id } })
   }
 
+  async deleteAllAssigneeTasks(assigneeId: string, assigneeType: AssigneeType) {
+    // Policies validation shouldn't be required here because token is from a webhook event
+    const tasks = await this.db.task.findMany({
+      where: { assigneeId, assigneeType, workspaceId: this.user.workspaceId },
+    })
+    if (!tasks.length) {
+      // If assignee doesn't have an associated task at all, skip logic
+      return
+    }
+    const labels = tasks.map((task) => task.label)
+
+    await this.db.$transaction([
+      this.db.task.deleteMany({
+        where: { assigneeId, assigneeType, workspaceId: this.user.workspaceId },
+      }),
+      this.db.label.deleteMany({ where: { label: { in: labels } } }),
```

Comment:
Are labels assignee specific?

Reply:
@sazanrjb 

Labels are computed like this:
- Rojan client doesn't have company -> ROJ-001
- Rojan client belongs to company Tesla -> TES-001
- Task assigned to tesla company itself -> TES-001
- Task assigned to IU Rojan belonging to Copilot + Outside workspace -> COP-001